File: app/src/main/java/com/anadi/weatherinfo/UpdateReceiver.java
package com.anadi.weatherinfo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import com.anadi.weatherinfo.addlocation.AddLocationContract;
import com.anadi.weatherinfo.addlocation.LocationsProvider;
import com.anadi.weatherinfo.repository.Country;
import com.anadi.weatherinfo.repository.Locations;
import com.anadi.weatherinfo.repository.LocationsCash;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class UpdateReceiver extends BroadcastReceiver {

    public static final String NOTIFICATION_ID = "notification-id";
    public static final String NOTIFICATION = "notification";
    private final LocationsProvider locationsProvider = Locations.getInstance();
    private final AddLocationContract.Model model = LocationsCash.getInstance();
    private final Executor exec = Executors.newSingleThreadExecutor();

    @Override
    public void onReceive(Context context, Intent intent) {
        //        String intentAction = intent.getAction();
        int id = intent.getIntExtra(NOTIFICATION_ID, 0);

        if (id == 0) {
            String toastMessage = "add random location";
            Toast.makeText(context, toastMessage, Toast.LENGTH_SHORT).show();

            Country country = locationsProvider.getRandomCountry();
            String cityName = locationsProvider.getRandomCity(country.getName());

            exec.execute(() -> {
                try {
                    boolean result = model.add(cityName, country.getName());

                } catch (Exception e) {
                    System.err.println(e.getMessage());
                    e.printStackTrace();
                }
            });

            return;
        }
    }
}

File: app/src/main/java/com/anadi/weatherinfo/UpdateReceiver.kt
package com.anadi.weatherinfo

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import com.anadi.weatherinfo.domain.location.AddLocationUseCase
import com.anadi.weatherinfo.utils.WeatherException
import es.dmoral.toasty.Toasty
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import timber.log.Timber
import javax.inject.Inject

class UpdateReceiver @Inject constructor(private val addLocationUseCase: AddLocationUseCase) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        //        String intentAction = intent.getAction();
        val id = intent.getIntExtra(NOTIFICATION_ID, 0)
        if (id == 0) {
            Toasty.info(context, "Add random location", Toast.LENGTH_SHORT).show()

            GlobalScope.launch {
                try {
                    addLocationUseCase.build(AddLocationUseCase.Params.random())
                } catch (e: WeatherException) {
                    Timber.e(e)
                }
            }
        }
    }

    companion object {
        const val NOTIFICATION_ID = "notification-id"
        const val NOTIFICATION = "notification"
    }
}